Resubmit sentence

Burge337 opened the pull request:
Fixes issue #38

Looking for feedback on whether we want kids to be able to place the re submitted sentence in between other word tiles in the sentence bar. As things stand right now, it just appends it to the end.

This branch also fixes an unlisted bug where clicking on word tiles in the sentence bar or the word holder wouldn't play TTS.

Burge337 committed:
Create setup for allowing users to resubmit user sentences

Currently copies and stores the word tiles from the sentence bar in the SentenceInTiles game object which will eventually "offload" those tiles into the sentence bar.

Also contains a placeholder image component on the new game object so dragging the tile is obvious.

Burge337 committed:
Implement re submission of completed sentences to sentence bar

The current implementation doesn't allow for positioning the resubmitted sentence in between words already in the sentence bar. It always appends to the end. If it turns out that we think it's important to be able to place the sentence in between word tiles, that can arranged.

This also fixes an unlisted issue where clicking on word tiles in the sentence bar and word holder wouldn't actually play the TTS. Turns out that when we'd added the voice modifiers, I'd forgotten to add the TTS handler component to them, and a script that thought it would have it was throwing a null reference exception. 

fixes #38

kklamberty reviewed (changes requested):
Let's make it so that the sentence looks more tile like (maybe like the draggable sentence form the story building view). Let's make sure the sentence is on the top while it's being dragged around.

Burge337 committed:
Create word tile image for the submitted sentences

I've also made it so that the sentence tile is above everything else, with the exception of the sentence view. Because of some funky behavior with eventData.hovered, the dragged object needs to be below the sentence view in order to actually detect it. This is probably because the dragged object blocks raycasts otherwise, but I'm still looking into it.

Burge337 committed:
Implement raycastTarget updates on drag method

This fixes the previous quirk of the system where eventData.hovered only worked when the object being dragged was underneath the sentence in the game view. The reason that was happening was because eventData.hovered was relying on raycasts to find things in the hovered stack that were being blocked by the dragged object.

Burge337 committed:
Re-work leantween animation to fit new sentence set up

Now lean tween animates the whole collection of parent and child objects rather than just the text component.

Also added a white UI image behind the pipes to mask the sentence tile animating in. (so it looks like the tile is coming out of the pipe rather than off screen)

Burge337 committed:
Fix issue with lever animation where aspect ratios affected positioning

The reveal animation for the sentence tile now uses positioning relative to its parent to ensure it doesn't end up in strange places when you mess with the aspect ratio. Also extended the white space a bit to fully block the space behind the pipe during the animation.

Burge337 committed:
Change the size of the grey border of the sentence tile

Previously it looked overly elliptical, this change makes it more in line with how the word tiles look.

Burge337 committed:
Merge branch 'master' into resubmit-sentence
